optimizely/decision/experiment_whitelist_service.go

```diff
@@ -0,0 +1,58 @@
+/****************************************************************************
+ * Copyright 2019, Optimizely, Inc. and contributors                        *
+ *                                                                          *
+ * Licensed under the Apache License, Version 2.0 (the "License");          *
+ * you may not use this file except in compliance with the License.         *
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+// Package decision //
+package decision
+
+import (
+	"errors"
+
+	"github.com/optimizely/go-sdk/optimizely/decision/reasons"
+	"github.com/optimizely/go-sdk/optimizely/entities"
+)
+
+// ExperimentWhitelistService makes a decision using an experiment's whitelist (a map of user id to variation keys)
```

Comment:
I don't think I did this in order classes, but would be good to mention in the docstring that it implements the `ExperimentService` interface